internal/provider/resource_application.go

```diff
@@ -373,6 +515,18 @@ func (n nestedEndpointBinding) transformToStringTuple() (string, string) {
 	return n.Endpoint.ValueString(), n.Space.ValueString()
 }
 
+// nestedStorage represents the single element of the storage SetNestedAttribute
+type nestedStorage struct {
+	Label types.String `tfsdk:"label"`
+	Size  types.String `tfsdk:"size"`
+	Pool  types.String `tfsdk:"pool"`
+	Count types.Int64  `tfsdk:"count"`
+}
+
+func (n nestedStorage) transformToString() string {
+	return fmt.Sprintf("%s=%s,%s,%d", n.Label.ValueString(), n.Size.ValueString(), n.Pool.ValueString(), n.Count.ValueInt64())
```

Comment:
This is a good place to use String() rather than ValueString(). It'll do the quoting for you. The alternative is to use `%q` rather than `%s`.